### PR TITLE
tensorflow-lite: add version 2.14.0

### PR DESCRIPTION
**Pull request by Wawha**
Library name and version:  **tensorflow-lite/2.14.0**

Add new version 2.14.0 of Tensorflow lite.
Modifications compare to tensorflow repo:
- Add by headers files of library `ml_dtypes` to be able to compile.

Conan package modification:
- Add Android support in `test_package`.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/tensorflow-lite/all/conandata.yml [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
     url: "https://github.com/tensorflow/tensorflow/archive/refs/tags/v2.10.0.tar.gz"
     sha256: "b5a1bb04c84b6fe1538377e5a1f649bb5d5f0b2e3625a3c526ff3a8af88633e8"
   "2.10.0":
     url: "https://github.com/tensorflow/tensorflow/archive/refs/tags/v2.10.0.tar.gz"
     sha256: "b5a1bb04c84b6fe1538377e5a1f649bb5d5f0b2e3625a3c526ff3a8af88633e8"
 patches:
+  "2.14.0":
+    - patch_file: "patches/disable_fetch_content.patch"
+      patch_description: "Fail if the CMake build script tries to fetch external dependencies"
+      patch_type: "conan"
```
Comment (valgur) [inline recipes/tensorflow-lite/all/conandata.yml:15]: This patch can probably be replaced with simply setting [`FETCHCONTENT_FULLY_DISCONNECTED`](https://cmake.org/cmake/help/latest/module/FetchContent.html#variable:FETCHCONTENT_FULLY_DISCONNECTED).

File: recipes/tensorflow-lite/all/patches/dependencies.patch [hunk 2]
```diff
@@ -0,0 +1,124 @@
+diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
+index 0924ca9dacd..fed88560a02 100644
+--- a/tensorflow/lite/CMakeLists.txt
++++ b/tensorflow/lite/CMakeLists.txt
+@@ -146,37 +146,16 @@ set(TFLITE_TARGET_DEPENDENCIES "")
+ find_package(absl REQUIRED)
+ find_package(Eigen3 REQUIRED)
+ find_package(farmhash REQUIRED)
+-find_package(fft2d REQUIRED)
++find_package(fft REQUIRED)
```
Comment (valgur) [inline recipes/tensorflow-lite/all/patches/dependencies.patch:10]: You can probably do `deps = ConanDeps(self); deps.set_property("fft", "cmake_file_name", "fft2d")` instead.

Comment (valgur) [reply recipes/tensorflow-lite/all/patches/dependencies.patch:10]: The same goes for xnnpack.

File: recipes/tensorflow-lite/all/patches/dependencies.patch [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+     absl::synchronization
+     absl::variant
+-    farmhash
++    farmhash::farmhash
+     fft2d_fftsg2d
+-    flatbuffers::flatbuffers
+-    gemmlowp::gemmlowp
+-    ml_dtypes
++    ${FLATBUFFERS_TARGET}
++    gemmlowp::eight_bit_int_gemm
```
Comment (valgur) [inline recipes/tensorflow-lite/all/patches/dependencies.patch:121]:
```
deps.set_property("farmhash", "cmake_target_name", "farmhash")
deps.set_property("gemmlowp", "cmake_target_name", "gemmlowp::gemmlowp")

File: recipes/tensorflow-lite/all/patches/ml_dtypes.patch
Comment (valgur) [inline recipes/tensorflow-lite/all/patches/ml_dtypes.patch:1]: At this point I would rather `download()` it from GitHub instead. That's a bit excessive for a patch.

**Review by valgur — CHANGES_REQUESTED** [review]
The full header file is a tad excessive for a patch. Patching can probably simplified slightly. Otherwise LGTM.

Thanks!

**Comment by Wawha** [comment]
> The full header file is a tad excessive for a patch. Patching can probably simplified slightly. Otherwise LGTM.
> 
> Thanks!

Thank you for the review. Unfortunately I'm not working anymore on that project, so I do not have the setup to test the modification I could make with you remarks.
Perhaps I should close the pull request

**Pull request closed without merging**